contrib/guix/guix-codesign

```diff
@@ -227,7 +227,7 @@ SOURCE_DATE_EPOCH="${SOURCE_DATE_EPOCH:-$(git log --format=%at -1)}"
 time-machine() {
     # shellcheck disable=SC2086
     guix time-machine --url=https://git.savannah.gnu.org/git/guix.git \
-                      --commit=aa34d4d28dfe25ba47d5800d05000fb7221788c0 \
+                      --commit=ae03f401381e956c4c41b4cf495cbde964fa43d0 \
```

Comment:
nit: should be done "Merge bitcoin/bitcoin#24484: guix: use same commit for codesigning time-machine" instead of this backport, because the bitcoin/bitcoin#24485 is a backport 24484 to bitcoin-v23.

Reply:
Sorry, I just noticed one more thing @PastaPastaPasta 

#24484 probably should be backported only after these changes:
 - Merge bitcoin/bitcoin#23778: release: Guix 1.4.0 & GCC 10.3
 - Merge bitcoin/bitcoin#24042: build: Point Guix to the current top of the "version-1.4.0" branch
 - Merge bitcoin/bitcoin#24057: guix: point to recent commit on the master branch
 - Merge bitcoin/bitcoin#24348: build: Fix Guix build for Windows (attempt 2)

because otherwise versions are mismatched in opposite way (`guix-codesign` is newer than `guix-build`).
But if it works - doesn't really matter.

Reply:
Agreed; dropped.